src/main/java/com/minecolonies/blockout/util/Parsing.java

```diff
@@ -0,0 +1,170 @@
+package com.minecolonies.blockout.util;
+
+import net.minecraft.util.math.MathHelper;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.minecolonies.blockout.util.Log.getLogger;
+
+/**
+ * A utility class with method that handle parsing tasks.
+ */
+public class Parsing
+{
+    private static final Pattern PERCENTAGE_PATTERN = Pattern.compile("([-+]?\\d+)(%|px)?", Pattern.CASE_INSENSITIVE);
+    private static final Pattern RGBA_PATTERN       =
+      Pattern.compile("rgba?\\(\\s*(\\d+)\\s*,\\s*(\\d+)\\s*,\\s*(\\d+)\\s*(?:,\\s*([01]\\.\\d+)\\s*)?\\)", Pattern.CASE_INSENSITIVE);
+
+    private Parsing() {}
+
+    /**
+     * Parse the scalable integer attribute from name and definition.
+     *
+     * @param attr  the attr.
+     * @param def   the definition.
+     * @param scale the scale.
+     * @return the integer.
+     */
+    public static int parseScalableInteger(final String attr, final int def, final int scale)
+    {
+        if (attr != null)
+        {
+            final Matcher m = PERCENTAGE_PATTERN.matcher(attr);
+            if (m.find())
+            {
+                return parseScalableIntegerUsingMatcher(m, def, scale);
+            }
+        }
+
+        return def;
+    }
+
+    /**
+     * Parses a scaled integer using the output of the {@link Matcher}.
+     *
+     * @param m     The matcher whos output is used.
+     * @param def   The default value.
+     * @param scale The maximal value.
+     * @return The scaled value.
+     */
+    public static int parseScalableIntegerUsingMatcher(final Matcher m, final int def, final int scale)
+    {
+        try
+        {
+            int value = Integer.parseInt(m.group(1));
+
+            if ("%".equals(m.group(2)))
+            {
+                value = scale * MathHelper.clamp(value, 0, 100) / 100;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 100 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 100 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)